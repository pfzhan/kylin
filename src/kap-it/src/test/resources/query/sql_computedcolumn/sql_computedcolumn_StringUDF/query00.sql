--
-- Copyright (C) 2020 Kyligence Inc. All rights reserved.
--
-- http://kyligence.io
--
-- This software is the confidential and proprietary information of
-- Kyligence Inc. ("Confidential Information"). You shall not disclose
-- such Confidential Information and shall use it only in accordance
-- with the terms of the license agreement you entered into with
-- Kyligence Inc.
--
-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-- "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-- LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-- A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-- OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-- LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-- DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-- THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
--


select
    COUNT(CONCAT("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")),
    count(substr(LSTG_FORMAT_NAME,2)),
    COUNT(INSTR("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")) "INSTRa",
    COUNT(INSTR("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME", 1)) "INSTRb",
    COUNT(LENGTH("LSTG_FORMAT_NAME")),
    COUNT(STRPOS("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")),
    count(DISTINCT substr(LSTG_FORMAT_NAME,2)),
    count(DISTINCT concat(LSTG_FORMAT_NAME,LSTG_FORMAT_NAME)),
    count(DISTINCT instr(LSTG_FORMAT_NAME,LSTG_FORMAT_NAME)) "INSTRc",
    count(DISTINCT length(LSTG_FORMAT_NAME))
from test_kylin_fact