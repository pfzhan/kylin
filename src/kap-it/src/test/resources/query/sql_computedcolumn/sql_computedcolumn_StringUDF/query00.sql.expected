

select
    COUNT(CONCAT("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")),
    count(substr(LSTG_FORMAT_NAME,2)),
    COUNT(INSTR("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")) "INSTRa",
    COUNT(INSTR("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")) "INSTRb",
    COUNT(LENGTH("LSTG_FORMAT_NAME")),
    COUNT(INSTR("LSTG_FORMAT_NAME", "LSTG_FORMAT_NAME")),
    count(DISTINCT substr(LSTG_FORMAT_NAME,2)),
    count(DISTINCT concat(LSTG_FORMAT_NAME,LSTG_FORMAT_NAME)),
    count(DISTINCT instr(LSTG_FORMAT_NAME,LSTG_FORMAT_NAME)) "INSTRc",
    count(DISTINCT length(LSTG_FORMAT_NAME))
from test_kylin_fact